### ISSUE TITLE
117 - Type was not found or was not a compile-time constant: BulkLoader 

### ISSUE THREAD

**Issue by arthur-debert** [arthur-debert/BulkLoader#116]
<pre>
<b>What steps will reproduce the problem?</b>
1. Trying to compile.
2. Type was not found or was not a compile-time constant: BulkLoader.

<b>What is the expected output? What do you see instead?</b>
I don't see anything, it won't load BulkLoader at all.

<b>What version of the product are you using? On what operating system?</b>
OSX Flash CS4, the latest build rev-282

</pre></br>Original link: http://code.google.com/p/bulk-loader/issues/detail?id=117

**Comment by arthur-debert** [comment]
gerry00creighton - 2010-04-10 00:00:00 

I have a number of other imports that work fine and are in the same directory as the
br directory. Is BulkLoader dependent on anything else? My main document class
extends papervision3D BasicView. I know of another dev that was having problems
yesterday, not sure what his were but it was a strange coincidence.

**Comment by arthur-debert** [comment]
debert - 2010-04-10 00:00:00 
None

**Comment by arthur-debert** [comment]
debert - 2010-04-10 00:00:00 

Hi Gerry.

I cannot reproduce this here. Besides, a bunch of other folks are using the zip file, and this is the first report 
of it's kind. This zip has over15 thousand downloads.

(BulkLoader's only dependeny is the corelib JSON encoder **if** you are using JSONItem )

My suggestion is the usual:
- Open a new fla.
- On first frame import BulkLoader + create a instance

What should happen if there was something wrong is you'd get an error.

**Comment by arthur-debert** [comment]
astropcr - 2010-06-11 00:00:00 

Sorry, couldn't find anything about the dependency anywhere else, would be good if there was a page talking about where to get it and how to set it up for BulkLoader.

But I assuming if I am not going to use a certain type of Bulk Loader I can just remove the class from the loadingtypes folder with no ill effects? For example, I can remove the JSONItem and everything else should be fine?

**Comment by arthur-debert** [comment]
lucks17 - 2010-05-14 00:00:00 
None

**Comment by arthur-debert** [comment]
astropcr - 2010-06-18 00:00:00 

It encountered the dependency problem when I tried to compile BulkLoader in a .swc so that I could just drop it into my project. I am guessing from what you said that when the Flex compiler makes a .swc it tries to grab all the dependencies, even if they are not done by default? 

I only removed the JSONItem and it compiled into a .swc with no problems. Obviously BulkLoader's functionality is more limited now, but there is enough there for me to use.

Thanks for the great code!
